File: sol/solution/0001-0100/0027. Remove Element/Solution.ts
function removeElement(nums: number[], val: number): number {
  let n = nums.length, i = 0;
  while (i < n) {
    if (nums[i] === val) {
      nums[i] = nums[n - 1];
      n--;
    } else {
      i++;
    }
  }
  return n;
}

File: sol/solution/0001-0100/0027. Remove Element/Solution.rb
# @param {Integer[]} nums
# @param {Integer} val
# @return {Integer}
def remove_element(nums, val)
  n = nums.length
  i = 0
  while i < n
    if nums[i] == val
      nums[i] = nums[n - 1]
      n -= 1
    else
      i += 1
    end
  end
  n
end